From c5e03333dc8d888736166e247022c0a3b53135d3 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Tue, 13 Jan 2026 12:18:00 -0600
Subject: [PATCH 1/1] Drop unnecessary scope qualifiers to please flake8

This variable is not assigned to, and so the scope qualifier is not
necessary and may be misleading.

Upstream-Status: Backport [c5e03333dc8d888736166e247022c0a3b53135d3]

---
 colcon_ros/package_augmentation/ros_ament_python.py | 2 --
 colcon_ros/package_identification/ros.py            | 2 --
 colcon_ros/prefix_path/ament.py                     | 1 -
 colcon_ros/prefix_path/catkin.py                    | 1 -
 test/test_spell_check.py                            | 1 -
 5 files changed, 7 deletions(-)

diff --git a/colcon_ros/package_augmentation/ros_ament_python.py b/colcon_ros/package_augmentation/ros_ament_python.py
index 532d5f8..c9a068c 100644
--- a/colcon_ros/package_augmentation/ros_ament_python.py
+++ b/colcon_ros/package_augmentation/ros_ament_python.py
@@ -46,14 +46,12 @@ class RosAmentPythonPackageAugmentation(PackageAugmentationExtensionPoint):
                         options = config.get('options', {})
 
                         def getter(env):
-                            nonlocal options
                             return options
                         break
         else:
             # use information from setup.py file
 
             def getter(env):  # noqa: F811
-                nonlocal desc
                 return get_setup_information(
                     desc.path / 'setup.py', env=env)
 
diff --git a/colcon_ros/package_identification/ros.py b/colcon_ros/package_identification/ros.py
index 4bfe190..ce68787 100644
--- a/colcon_ros/package_identification/ros.py
+++ b/colcon_ros/package_identification/ros.py
@@ -86,7 +86,6 @@ class RosPackageIdentification(
             descs, additional_argument_names=additional_argument_names)
 
         # get all parsed ROS package manifests
-        global _cached_packages
         pkgs = {}
         for desc in descs:
             if str(desc.path) not in _cached_packages:
@@ -161,7 +160,6 @@ def add_group_dependencies(pkgs):
 
 def get_package_with_build_type(path: str):
     """Get the ROS package and its build type for the given path."""
-    global _cached_packages
     if path not in _cached_packages:
         pkg = _get_package(path)
         build_type = _get_build_type(pkg, path) if pkg else None
diff --git a/colcon_ros/prefix_path/ament.py b/colcon_ros/prefix_path/ament.py
index e060c9c..388ece6 100644
--- a/colcon_ros/prefix_path/ament.py
+++ b/colcon_ros/prefix_path/ament.py
@@ -24,7 +24,6 @@ class AmentPrefixPath(PrefixPathExtensionPoint):
             PrefixPathExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
 
     def extend_prefix_path(self, paths):  # noqa: D102
-        global _get_ament_prefix_path_warnings
         ament_prefix_path = os.environ.get('AMENT_PREFIX_PATH', '')
         for path in ament_prefix_path.split(os.pathsep):
             if not path:
diff --git a/colcon_ros/prefix_path/catkin.py b/colcon_ros/prefix_path/catkin.py
index 624e404..260f3b1 100644
--- a/colcon_ros/prefix_path/catkin.py
+++ b/colcon_ros/prefix_path/catkin.py
@@ -28,7 +28,6 @@ class CmakePrefixPath(PrefixPathExtensionPoint):
             PrefixPathExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
 
     def extend_prefix_path(self, paths):  # noqa: D102
-        global _get_cmake_prefix_path_warnings
         cmake_prefix_path = os.environ.get('CMAKE_PREFIX_PATH', '')
         for path in cmake_prefix_path.split(os.pathsep):
             if not path:
diff --git a/test/test_spell_check.py b/test/test_spell_check.py
index 5861814..39484e9 100644
--- a/test/test_spell_check.py
+++ b/test/test_spell_check.py
@@ -10,7 +10,6 @@ spell_check_words_path = Path(__file__).parent / 'spell_check.words'
 
 @pytest.fixture(scope='module')
 def known_words():
-    global spell_check_words_path
     return spell_check_words_path.read_text().splitlines()
 
 
-- 
2.47.3

